src/xenia/kernel/xboxkrnl/xboxkrnl_crypt.cc

```diff
@@ -216,19 +222,137 @@ void XeCryptSha256Final(pointer_t<XECRYPT_SHA256_STATE> sha_state,
 }
 DECLARE_XBOXKRNL_EXPORT1(XeCryptSha256Final, kNone, kImplemented);
 
-// Byteswap?
-dword_result_t XeCryptBnQw_SwapDwQwLeBe(lpqword_t qw_inp, lpqword_t qw_out,
-                                        dword_t size) {
-  return 0;
+// Byteswaps each 8 bytes
+void XeCryptBnQw_SwapDwQwLeBe(pointer_t<uint64_t> qw_inp,
+                              pointer_t<uint64_t> qw_out, dword_t size) {
+  for (uint32_t i = 0; i < size; i++) {
+    qw_out[i] = xe::byte_swap(qw_inp[i]);
+  }
 }
-DECLARE_XBOXKRNL_EXPORT1(XeCryptBnQw_SwapDwQwLeBe, kNone, kStub);
+DECLARE_XBOXKRNL_EXPORT1(XeCryptBnQw_SwapDwQwLeBe, kNone, kImplemented);
 
-dword_result_t XeCryptBnQwNeRsaPubCrypt(lpqword_t qw_a, lpqword_t qw_b,
-                                        lpvoid_t rsa) {
+typedef struct {
+  xe::be<uint32_t> size;  // size of modulus in 8 byte units
+  uint32_t public_exponent;
```

Comment:
Should probably be `xe::be<uint32_t>` for consistency (especially when other implementations are added), with byte swap explicitly bypassed (or done twice). The rest looks pretty good.

Reply:
Yeah true, looks like I can use .value to get the unswapped value too, updated it with that instead